### PR TITLE
Et sm empty list welcome

### LINKED ISSUE #8: 8. As a user, I want to see a welcoming prompt to add my first item if my list is empty to help me get oriented to how the app works
## Summary

When a user starts a new list, they need a little guidance on how to get started. Think through what you'd want to see as a user and make some intentional user experience decisions that will guide our users in confidently adding their very first item. 

## Acceptance criteria
- [x] The list view, when there are no items to display, should show a prompt (e.g., a button) for the user to add their first item

### PR DESCRIPTION
**Pull request by sdmisra**
## Description

This PR sets up a welcome message to the user that reflects whether or not the list they are viewing is empty or not, as well as checks whether or not a user has yet selected a list before navigating to the List view.

## Related Issue

Closes #8 

## Acceptance Criteria

- [x] The list view, when there are no items to display, should show a prompt (e.g., a button) for the user to add their first item

## Type of Changes

`Feature`

### Before

Screenshot 2024-02-27 at 8 11 16 AM

### After

Logged in user - after navigating to list that is empty:

Screenshot 2024-02-27 at 8 15 25 AM

Added prompt to select list (and button to go home) when user skips to List view without clicking list on Home:

Screenshot 2024-02-27 at 8 12 30 AM

## Testing Steps / QA Criteria

Testing for AC / empty list prompt:

1. Navigate to test deployment below, sign in
2. To avoid bugs, clear local storage after navigating to test deployment.
3. Make a new list by using form on Home view - this should redirect user to List view 
4. Confirm message indicates that new list is empty, click button to continue
5. Confirm that button redirects user to ManageList view

Testing for edge case - user has not selected a list before navigating to List view:
1. Navigate to test deployment below, sign in
2. Clear local storage
3. Navigate using buttons on bottom of screen to List view (without clicking anything but sign in beforehand)
4. Confirm message indicates that user has not selected list
5. Click button to confirm it redirects to Home view to enable list selection

### CODE REVIEW

File: src/views/List.jsx [hunk 1]
```diff
@@ -13,29 +16,50 @@ export function List({ data }) {
...
-			</form>
+			{!listPath && (
+				<>
+					<h2>
+						You haven't selected a list yet. Click below to select a list.
+					</h2>
+					<button onClick={() => handleClick('/')}>Select a list</button>
+				</>
+			)}
+			{listPath && data.length === 0 && (
```
Comment (sdmisra) [inline src/views/List.jsx]: Question about best practices for this sort of expression:

would ln 38 be better written as `{(listPath && data.length === 0) && (` ?

It seems to work just fine in the format it is in, but it sort of seems like it would be 'safer' to use parenthesis to make sure it is always evaluated in the order of operations that suit our needs.

Comment (luisaugusto) [reply src/views/List.jsx]: Hi @sdmisra, the parenthesis are not needed for this expression, but it's usually best to render conditional UI using ternaries instead of conditional chaining like this. So instead of `{listPath && data.length === 0 && <Component />}`, using `{listPath && data.length === 0 ? <Component /> : null}` instead. This [ESLint rule](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-no-leaked-render.md) explains why it's best to use the second approach.

Comment (sdmisra) [reply src/views/List.jsx]: @luisaugusto That makes sense - I can make that change to List.jsx and add it to this PR. Thank you for the clarification!

**Review by andiedoescode — DISMISSED** [review]
Looks good! I like the UX choice you made to have a button to redirect the user back to a list if they haven't selected one!

**Comment by DevinaG007** [comment]
Looks fantastic! I love your use of conditional rendering to display certain prompts to the user.

image

**Review by DevinaG007 — DISMISSED** [review]
Looks fantastic! I love your use of conditional rendering to display certain prompts to the user.

image

**Review by luisaugusto — APPROVED** [review]
Great work this week!

**Review by DevinaG007 — APPROVED** [review]
New ternaries look good!
Happy Dance Gif

**Review by andiedoescode — APPROVED** [review]
Still looks good!
